### DISCUSSION TITLE
Get url to continue cart session

### DISCUSSION THREAD

**Discussion by Web10-Joris** [duncanmcclean/simple-commerce#1032] [Help] [answered]
It would be incredibly useful to be able to generate a link from a particular cart in order to resume the session with that shopping cart. This link, for example, could be used to easily retrieve an abandonent cart. One click and the customer has their old cart back and can continue with the checkout. 
Another option might be to generate a link that creates a new cart and adds the same items to this cart and link it to the same customer. 

I am not sure if this is even possible, and maybe already possible somehow? Otherwise a nice feature request :D

**Comment by duncanmcclean** [comment, marked as answer]
Simple Commerce doesn't have anything built-in for this. However, you could achieve this with a custom route.

Assuming you're using the cookie cart driver, you could do something like this:

```php
// routes/web.php

use DuncanMcClean\SimpleCommerce\Facades\Order;
use Illuminate\Http\Request;
use Illuminate\Support\Facades\Cookie;
use Illuminate\Support\Facades\Route;
use Statamic\Facades\Blink;
use Statamic\Facades\Site;

Route::get('/back-to-cart/{orderId}', function (Request $request, $orderId) {
    $order = Order::find($orderId);

    if (! $order) {
        return redirect('/');
    }

    $key = config('simple-commerce.cart.key', 'simple-commerce-cart');

    $site = $this->guessSiteFromRequest();

    if (Site::hasMultiple() && ! config('simple-commerce.cart.single_cart')) {
        $key = $key.'-'.$site->handle();
    }

    Cookie::queue($key, $order->id);
    Blink::put($key, $order->id);

    return redirect('/cart');
});
```

Then, customers would be able to resume their cart by going to `/back-to-cart/the-order-id`

**Comment by Web10-Joris** [comment]
Thanks for this code example, works great! 

The method `$this->guessSiteFromRequest();` was not working inside this route context, therefore the  `artisan route:cache` resulted in a failure. I've fixed this by creating a controller for the route and including a copy of this function. 

Is it also possible to redirect the customer to the /cart on the same site as the order was originally created? For example the order is created on example.de, and the route example.com/back-to-cart/1234 will redirect to example.de/cart instead of example.com/cart.

**Reply by duncanmcclean** [reply]
> The method $this->guessSiteFromRequest(); was not working inside this route context, therefore the artisan route:cache resulted in a failure. I've fixed this by creating a controller for the route and including a copy of this function.

Whoops, I copied that bit in from the `CookieDriver` at the last moment before posting.

> Is it also possible to redirect the customer to the /cart on the same site as the order was originally created? For example the order is created on example.de, and the route example.com/back-to-cart/1234 will redirect to example.de/cart instead of example.com/cart.

Sure, you could do something like this:

```php
return redirect($order->resource()->site()->url() . '/cart');
```
